src/macros.jl

```diff
@@ -1490,6 +1490,42 @@ function build_variable(
     return VariableConstrainedOnCreation(variable, set)
 end
 
+function build_variable(
+    _error::Function,
+    variables::AbstractArray{<:ScalarVariable},
+    sets::AbstractArray{<:MOI.AbstractScalarSet},
+)
+    if length(variables) != length(sets)
```

Comment:
Here, if they don't have the same size, e.g. 2,2 and 4, the error will not catch the case

Reply:
I think Joaquim did this because `size` is not implemented for a `SparseAxisArray`. 

If they're regular arrays, Julia's broadcasting will catch it
```Julia
julia> x = [1 2; 3 4]
2×2 Matrix{Int64}:
 1  2
 3  4

julia> y = [5, 6, 7, 8]
4-element Vector{Int64}:
 5
 6
 7
 8

julia> f(x, y) = (x, y)
f (generic function with 1 method)

julia> f.(x, y)
ERROR: DimensionMismatch("arrays could not be broadcast to a common size; got a dimension with lengths 2 and 4")
```

Reply:
Ok, makes sense